### PR TITLE
Fixes to VIP plugins

### LINKED ISSUE #38: Better 'Community' plugin handling
Right now we've duct-taped the regular plugins list table into the VIP Plugins UI. This is not great because:
- Activate/deactivate still goes through `wp-admin/plugins.php`, so you end up on the wrong page
- Includes Bulk Actions
- UI does not support loading regular plugins via code - if a plugin has been loaded via code, the UI needs to display that message

We should make a custom list table for regular plugins in the same way we did for shared plugins, and use that here. Currently, we're using the core plugins table, crammed into our VIP plugins page.

That would include replicating some functionality from `wp-admin/plugins.php` for activation/deactivation, etc.

### LINKED ISSUE #43: Don't load UI enabled plugins from inside the Plugins UI class
Something called Plugins UI should _only_ handle Plugins UI :)

Really confusing that it handles loading the plugins also. Means it needs to be included from the frontend on every request, which is def. sub-optimal.

Refactor that functionality out. Really, it needs to be in mu-plugins directly, not part of VIP Dashboard either.

### LINKED ISSUE #45: Various removed shared plugins are referenced
`WPCOM_VIP_Plugins_UI::setup_globals` references various shared plugins which we've removed, these should no longer be referenced.

Hopefully a simple removal job. :smile:

### LINKED ISSUE #48: Activating a plugin with a fatal error is not possible and returns no error
When someone attempts to activate a plugin containing a fatal error in vanilla WordPress, the plugin is not activated and the UI displays an error message explaining why the plugin wasn't activated.

When someone attempts to activate a plugin containing a fatal error in VIP Go, the plugin is not activated but no error message is shown.

An error message should be displayed as it is in vanilla WordPress.

### LINKED ISSUE #49: Plugins activated via wpcom_vip_load_plugin do not appear activated in the UI
To reproduce:
1. Add a plugin, e.g. Field Manager, to the `plugins` directory on your repository
2. Activate the plugin via code, e.g. `wpcom_vip_load_plugin( 'fieldmanager/fieldmanager.php' );`
3. Visit the WP Admin > Plugins screen

Expected: Field Manager should show as activated with a message "Enabled via your theme's code"

Actual: Field Manager shows as inactive, BUT is actually active

### LINKED ISSUE #52: Switch back to core plugins screen
With the release of shiny updates (https://wordpress.org/plugins/shiny-updates/) in core, we should switch back to the standard plugins screen and find a different way to feature our partner plugins. Perhaps just a nice admin page or area at the top of the current plugins search screen.

### PR DESCRIPTION
**Pull request by scottsweb**
Being tracked here: https://github.com/Automattic/vip-go-mu-plugins/issues/206

Closes #52 
Closes #48 
Closes #45 
Closes #38 
Closes #43 
Closes #49 

screen shot 2016-06-07 at 20 45 35

screen shot 2016-06-07 at 20 45 59

### CODE REVIEW

**Comment by scottsweb** [comment]
May also close #43

File: plugins-ui/class.wpcom-vip-plugins-ui.php [hunk 1]
```diff
@@ -1,656 +0,0 @@
...
-		add_action( 'admin_post_' . self::ACTION_PLUGIN_ACTIVATE, array( $this, 'action_admin_post_plugin_activate' ) );
-		add_action( 'admin_post_' . self::ACTION_PLUGIN_DEACTIVATE, array( $this, 'action_admin_post_plugin_deactivate' ) );
-
-		add_action( 'admin_enqueue_scripts', array( $this, 'action_enqueue_scripts' ) );
-	}
-
-	/**
-	 * Includes any active plugin files that are enabled via the UI/option.
-	 */
-	public function include_active_plugins() {
```
Comment (nickdaugherty) [inline plugins-ui/class.wpcom-vip-plugins-ui.php]: It's super lame, but this is where plugins that have been enabled via UI get included, so we need to maintain this for sites that have already enabled plugins via the UI - even if we are ending the ability to do so with this change.

**Comment by scottsweb** [comment]
Fixed this @nickdaugherty, would love a second review.

Would be good to work out a strategy for testing this on some live sites too.

File: vip-dashboard.php [hunk 2]
```diff
@@ -365,3 +376,224 @@ function wpcom_vip_menu_order( $menu_ord ) {
...
+		foreach ( $fpp_plugins as $slug => $plugin) {
+			$image_src = plugins_url( 'assets/img/featured-plugins/' . $slug . '-2x.png', __DIR__ . '/vip-dashboard.php' );
+			$lobby_url = '//vip.wordpress.com/plugins/' . $slug . '/';
+			$is_active = vip_dashboard_is_plugin_active( $slug );
+		?>
+			<div class="plugin <?php if ( $is_active ) { ?>active<?php } ?>">
+				<img src="<?php echo esc_url( $image_src ); ?>" width="48" height="48" class="fp-icon" />
+				<div class="fp-content">
+					<h3 class="fp-title"><?php echo $plugin['name']; ?></h3>
+					<p class="fp-description"><?php echo $plugin['description']; ?></p>
```
Comment (nickdaugherty) [inline vip-dashboard.php]: These two lines need to be escaped.

File: vip-dashboard.php [hunk 3]
```diff
@@ -365,3 +376,224 @@ function wpcom_vip_menu_order( $menu_ord ) {
...
+				</div>
+				<div class="interstitial">
+					<div class="interstitial-inner">
+						<h3 class="fp-title"><?php echo $plugin['name']; ?></h3>
```
Comment (nickdaugherty) [inline vip-dashboard.php]: This line needs escaping.

**Comment by nickdaugherty** [comment]
@scottsweb I left a few notes about escaping.

Question - what happens to sites that already have plugins enabled via the UI? Will they be able to disable them after the changes here?

**Comment by scottsweb** [comment]
Escaping fixed.

> what happens to sites that already have plugins enabled via the UI? Will they be able to disable them after the changes here?

They will be stuck on... which is a problem for sure. We will need to work with each client to remove the plugins they have enabled in this way. We can then delete the option when the plugins are committed to the repo.

**Comment by nickdaugherty** [comment]
> We will need to work with each client to remove the plugins they have enabled in this way.

Alright - I think we should wait until that is completed before shipping this, otherwise we're setting ourselves up for frustration.

**Pull request closed without merging**